Refactor serde feature dependencies in Cargo.toml

We currently see this [build error](https://github.com/unicode-org/icu4x/pull/172/checks?check_run_id=1067514383#step:7:111) in PR #172 :

```
    Building crate=icu-pluralrules features=[serde]
...
error[E0433]: failed to resolve: use of undeclared type or module `tuple_vec_map`
  --> components\pluralrules\src\data\cldr_resource.rs:43:38
   |
43 | #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
   |                                      ^^^^^^^^^ use of undeclared type or module `tuple_vec_map`

error[E0433]: failed to resolve: use of undeclared type or module `tuple_vec_map`
```

As discussed in the meeting, it may be solved by refactoring the feature dependency declarations in Cargo.toml.

--- a/components/pluralrules/src/data/cldr_resource.rs
+++ b/components/pluralrules/src/data/cldr_resource.rs
@@ -1,27 +1,27 @@
 use crate::PluralCategory;
 
-#[cfg(feature = "serde")]
+#[cfg(feature = "io")]
 use serde::{Deserialize, Serialize};
 
 #[derive(PartialEq, Debug)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "io", derive(Serialize, Deserialize))]
 pub struct Resource<'s> {
-    #[cfg_attr(feature = "serde", serde(borrow))]
+    #[cfg_attr(feature = "io", serde(borrow))]
     pub supplemental: Supplemental<'s>,
 }
 
 #[derive(PartialEq, Debug)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "io", derive(Serialize, Deserialize))]
 pub struct LocalePluralRules<'s> {
-    #[cfg_attr(feature = "serde", serde(rename = "pluralRule-count-zero"))]
+    #[cfg_attr(feature = "io", serde(rename = "pluralRule-count-zero"))]
     pub zero: Option<&'s str>,
-    #[cfg_attr(feature = "serde", serde(rename = "pluralRule-count-one"))]
+    #[cfg_attr(feature = "io", serde(rename = "pluralRule-count-one"))]
     pub one: Option<&'s str>,
-    #[cfg_attr(feature = "serde", serde(rename = "pluralRule-count-two"))]
+    #[cfg_attr(feature = "io", serde(rename = "pluralRule-count-two"))]
     pub two: Option<&'s str>,
-    #[cfg_attr(feature = "serde", serde(rename = "pluralRule-count-few"))]
+    #[cfg_attr(feature = "io", serde(rename = "pluralRule-count-few"))]
     pub few: Option<&'s str>,
-    #[cfg_attr(feature = "serde", serde(rename = "pluralRule-count-many"))]
+    #[cfg_attr(feature = "io", serde(rename = "pluralRule-count-many"))]
     pub many: Option<&'s str>,
 }
 
@@ -40,20 +40,20 @@ impl<'s> LocalePluralRules<'s> {
 }
 
 #[derive(PartialEq, Debug)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "io", derive(Serialize, Deserialize))]
 pub struct Rules<'s>(
-    #[cfg_attr(feature = "serde", serde(with = "tuple_vec_map"))]
-    #[cfg_attr(feature = "serde", serde(borrow))]
+    #[cfg_attr(feature = "io", serde(with = "tuple_vec_map"))]
+    #[cfg_attr(feature = "io", serde(borrow))]
     pub Vec<(&'s str, LocalePluralRules<'s>)>,
 );
 
 #[derive(PartialEq, Debug)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "io", derive(Serialize, Deserialize))]
 pub struct Supplemental<'s> {
-    #[cfg_attr(feature = "serde", serde(rename = "plurals-type-cardinal"))]
-    #[cfg_attr(feature = "serde", serde(borrow))]
+    #[cfg_attr(feature = "io", serde(rename = "plurals-type-cardinal"))]
+    #[cfg_attr(feature = "io", serde(borrow))]
     pub plurals_type_cardinal: Option<Rules<'s>>,
-    #[cfg_attr(feature = "serde", serde(rename = "plurals-type-ordinal"))]
-    #[cfg_attr(feature = "serde", serde(borrow))]
+    #[cfg_attr(feature = "io", serde(rename = "plurals-type-ordinal"))]
+    #[cfg_attr(feature = "io", serde(borrow))]
     pub plurals_type_ordinal: Option<Rules<'s>>,
 }

